repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/gatherinfo/GatherTestInformation.java

```diff
@@ -194,23 +197,23 @@ public FileVisitResult preVisitDirectory(Path file, BasicFileAttributes attrs) t
         }
 
         if (!this.skipSettingStatusInformation) {
-            this.writeProperty("error-types", this.failureNames);
-            this.writeProperty("failing-test-cases", this.failureLocations);
-            this.writeProperty("totalNumberFailingTests", this.nbFailingTests);
-            this.writeProperty("totalNumberErroringTests", this.nbErroringTests);
-            this.writeProperty("totalNumberSkippingTests", this.nbSkippingTests);
-            this.writeProperty("totalNumberRunningTests", this.nbTotalTests);
-            this.getInspector().getJobStatus().setFailureLocations(this.failureLocations);
-
-            Metrics metrics = this.getInspector().getJobStatus().getMetrics();
+            jobStatus.writeProperty("error-types", this.failureNames);
```

Comment:
Same comment as below: we could move the write properties in the setters of JobStatus.

Reply:
For the comment below, I agree, but here I don't really understand, since `JobStatus` does not have setters for these stuff. Is supposed for me to create them?

Reply:
No actually I was saying that because they are in metrics. But for now leave it as it is, add just a todo comment to fix that later.

Reply:
Just for the record, actually I think that this `writeProperty` method will disappear in the future when `repairnator.json` is refactored to close to what `bears.json` is, because for `bears.json` we don't use this method (everything is in `Metrics4Bears`).

Reply:
Do you want I remove the duplication instead of putting the TODO comment? If yes, what do I remove? `jobStatus.writeProperty` or `metrics.set`? See that only for `failureLocations` we need the `jobStatus.writeProperty`, because apparently this property doesn't exist in `Metrics`.

Reply:
yes keep the failure location remove the rest

Reply:
Ok.